test/ads.jl

```diff
@@ -0,0 +1,129 @@
+using TaylorIntegration
+using Test
+
+@testset "Automatic Domain Splitting" begin
+    using TaylorIntegration: ADSBinaryNode, countnodes, timesvector
+    using StaticArrays: SVector
+
+    # This example is based upon section 3 of
+    # https://doi.org/10.1007/s10569-015-9618-3.
+
+    # Dynamical function
+    @taylorize function kepler_eqs!(dq, q, params, t)
```

Comment:
I suggest adding the following check in the Kepler problem: Say we take an initial (non-nominal) condition `q1` in the initial domain defined by `q0` and `dom`, which will have initial Keplerian elements say `a1`, `e1` (semimajor axis and eccentricity, etc.). Then, at the end of the ADS integration, the propagated initial condition `q2` will have a semimajor axis `a2`, eccentricity `e2`, etc. And we should have that `a1` and `a2` (resp., `e1` and `e2`) are equal within the order of magnitude of the splitting tolerance `stol`.

Reply:
Another check I think worth adding: take a number (say, 16) of initial conditions in the border of the domain (e.g., in the corners), propagate them individually without ADS, then check the difference at the end of the integration wrt ADS propagation.